modules/ding_webtrekk/ding_webtrekk.module

```diff
@@ -90,35 +153,170 @@ function ding_webtrekk_views_post_execute(&$view) {
     $loaded = TRUE;
 
     $params = [
-      'internalSearch' => array_shift($view->args),
-      'numberSearchResults' => $view->total_rows,
-      'pageTitle' => 'Search content',
+      'OSS' => array_shift($view->args),
+      'OSSr' => $view->total_rows,
     ];
 
-    _ding_webtrekk_set_data_layer_values($params);
+    // If using views pager; track the current search result page here also.
+    if (isset($view->query->pager->current_page)) {
+      $params['p_s_Page'] = ++$view->query->pager->current_page;
+    }
+
+    ding_webtrekk_add_page_parameters($params);
   }
 }
 
 /**
- * Pass Tag Information variables to the inline js script.
+ * Implements hook_ding_entity_view().
+ */
+function ding_webtrekk_ding_entity_view($entity, $view_mode) {
+  $entity_path = entity_uri('ting_object', $entity)['path'];
+
+  // The page parametes for the ting object view should only be added on the
+  // full page view for the material.
+  if ($view_mode == 'full' && current_path() == $entity_path) {
+    $params = [
+      'p_mat_type' => $entity->getType(),
+      'p_mat_indexno' => $entity->getClassification(),
+      'p_mat_lang' => $entity->getLanguage(),
+      'p_mat_source' => $entity->getAc_source(),
+      'p_mat_category' => implode(';', $entity->reply->getGenre()),
+    ];
+    ding_webtrekk_add_page_parameters($params);
+  }
+
+  // Look for at reserve button on this ding entity and setup a Webtrekk event
+  // if present. These could be added in every view_mode and page.
+  if (isset($entity->content['ding_entity_buttons'])) {
+    foreach (element_children($entity->content['ding_entity_buttons']) as $delta) {
+      foreach ($entity->content['ding_entity_buttons'][$delta] as $key => &$button) {
+        // Reserve button.
+        if (isset($button['#path']) && $button['#path'] == $entity_path . '/reserve') {
+          $button['#options']['attributes']['class'][] = 'ding-webtrekk-event';
+          $event_data = json_encode([
+            'linkId' => 'Reserver',
+            'customClickParameter' => [50 => $entity->getId()],
+          ]);
+          $button['#options']['attributes']['data-ding-webtrekk-event'] = $event_data;
```

Comment:
Perhaps it is worth implementing a shared method for adding webtreck data to an element? It seems to occur multiple places throughout the code.

Example:

```php
function _ding_webtrekk_add_data(&$element, $link_id, $data) {
  $event_data = [
            'linkId' => $link_id,
            'customClickParameter' => $data,
  ];
  $element['#attributes']['class'][] = 'ding-webtrekk-event';
  $element'#attributes']['data-ding-webtrekk-event'] = json_encode($event_data);
}
```

Reply:
Good idea. I will implement this.